src/NServiceBus.Persistence.DynamoDB/SynchronizedStorage/SynchronizedStorageSessionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using Persistence;
+    using Persistence.DynamoDB;
+
+    /// <summary>
+    /// DynamoDB persistence specific extension methods for the <see cref="ISynchronizedStorageSession"/>.
+    /// </summary>
+    public static class SynchronizedStorageSessionExtensions
+    {
+        /// <summary>
+        /// Retrieves the shared <see cref="IDynamoDBStorageSession"/> from the <see cref="ISynchronizedStorageSession"/>.
+        /// </summary>
+        public static IDynamoDBStorageSession DynamoDBPersistenceSession(this ISynchronizedStorageSession session)
+        {
+            Guard.AgainstNull(nameof(session), session);
+
+            if (session is not IDynamoDBStorageSession dynamoSession)
+            {
+                throw new Exception($"Cannot access the synchronized storage session. Ensure that 'EndpointConfiguration.UsePersistence<{nameof(DynamoDBPersistence)}>()' has been called.");
+            }
```

Comment:
Shall this be more explicit? for example, if `session` is `null` the message might say "no persistence" configured. If it's not null and is not of the expected type, then it can say "are you using a different persister?"

Reply:
You mean something like? https://github.com/Particular/NServiceBus.Persistence.Sql/blob/master/src/SqlPersistence/SqlPersistenceStorageSessionExtensions.cs#L29

Reply:
Pushed